### DISCUSSION TITLE
How to convert a vector if Arrays into an Array along a new dimension?

### DISCUSSION THREAD

**Discussion by zgbkdlm** [rust-ndarray/ndarray#1150] [Q&A] [answered]
Suppose that we have a vector of 5 Arrays. Also let the shapes of theses Arrays be uniform and known, say (3, 3). 

```rust
let a: Vec<Array2<f64>> = vec![ # five Arrays ];
```

Do we have any constructor method to create a new Array of shape (5, 3, 3) from this vector?

**Comment by jturner314** [comment, marked as answer]
This is an area which could be improved. There are a couple of ways to do this using existing functionality:

```rust
use ndarray::prelude::*;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let owned: Vec<Array2<f64>> = vec![Array2::zeros([3, 3]), Array2::ones([3, 3])];
    
    // Option 1:
    let views: Vec<ArrayView2<'_, f64>> = owned.iter().map(|arr| arr.view()).collect();
    let stacked = ndarray::stack(Axis(0), &views)?;
    println!("Option 1:\n{stacked}\n");
    
    // Option 2:
    let mut stacked: Array3<f64> = Array3::zeros([0, 3, 3]);
    for arr in &owned {
        stacked.push(Axis(0), arr.view())?;
    }
    println!("Option 2:\n{stacked}\n");
    
    Ok(())
}
```

([Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=7724392c672246e223f6a4256ce780d8))

**Reply by marstaik** [reply]
Is it possible to have a stack of views along an axis produce a non-owned ArrayView instead of an ArrayBase?
So the data would look like:
```
[
 &[[1,2,3],
     [4,5,6],
     [7,8,9]],
 &[[9,8,7],
     [6,5,4],
     [3,2,1]]
 ]

**Reply by nilgoyette** [reply]
This question doesn't seem to be related to the current discussion...?

See this [issue](https://github.com/rust-ndarray/ndarray/issues/1338). The answer is no because each view is allocated randomly in memory and there's no way that this new ArrayView could make sense of it. It must own the data.
